selfdrive/controls/controlsd.py

```diff
@@ -166,6 +167,11 @@ def update_events(self, CS):
     if self.sm['thermal'].memUsedPercent > 90:
       self.events.add(EventName.lowMemory)
 
+    # Check if all manager processes are running
+    not_running = set(p.name for p in self.sm['managerState'].processes if not p.running)
+    if self.sm.rcv_frame['managerState'] and (not_running - IGNORE_PROCESSES):
```

Comment:
hmm, why don't we add a timeout to rcv_frame? it's possible for this check to not work if the packet is never published

Reply:
guess the alive check will catch this